test/validation-error-handling.test.js

```diff
@@ -180,3 +180,33 @@ test('Attached validation error should take precendence over setErrorHandler', t
     t.strictEqual(res.statusCode, 400)
   })
 })
+
+test('should handle response validation error', t => {
+  t.plan(2)
+
+  const response = {
+    200: {
+      type: 'object',
+      required: ['name', 'work'],
+      properties: {
+        name: { type: 'string' },
+        work: { type: 'string' }
+      }
+    }
+  }
+
+  const fastify = Fastify()
+
+  fastify.get('/', { schema: { response } }, function (req, reply) {
+    reply.code(200).send({ work: 'actor' })
```

Comment:
I'm not sure I like this, in my opinion `send` should never throw. Otherwise, we are "suggesting" users to always put `send` inside a try-catch block.

Reply:
@delvedor this is a philosophical decision 😄 We can hide the error and handle it with onErrorHook (see my first commit) or we can makes the error bubble up (my second commit). 
**1st approach**
We wouldn't handle a developer error, like @mcollina say, so Fastify shouldn't hide it. This isn't a common situation and surely caused by a bug. It have to be fixed by developer and not hidden  try-catching everything.
**2nd approach**
By the way, a big project has surely a good amount of bugs, and it's not nice that app crashes due to a missing field. A 500 error is really welcome in production. That's my case, I find this bug due to a malformed record in my MongoDB that caused the validation failure.

What if we let developer decide? With a validator option or something similar. I'm just saying, I'm not sure this match the project philosophy.

Reply:
I had a chant with @mcollina, and I agree with him.
We must be consistent with our current behavior if you are using a standard function as handler `reply.send` should throw and you should handle the exception, while if you are using an *async* handler, we should handle the exception internally.

Reply:
an async handler will catch the exception, we have to do nothing about it, just throw.